airflow/models/taskinstance.py

```diff
@@ -1265,6 +1558,132 @@ def _update_rtif(ti, rendered_fields, session: Session | None = None):
     RenderedTaskInstanceFields.delete_old_records(ti.task_id, ti.dag_id, session=session)
 
 
+def _coalesce_to_orm_ti(*, ti: TaskInstancePydantic | TaskInstance, session: Session):
```

Comment:
Can we ensure with typing?
```suggestion
def _coalesce_to_orm_ti(*, ti: TaskInstancePydantic | TaskInstance, session: Session) -> TaskInstance:
```